### ISSUE TITLE
Regarding Performance Degradation During the Training Process

### ISSUE THREAD

**Issue by YCZ55** [hebing-sjtu/SurfSplat#9]
Hi, 

I trained a model using an identical configuration—specifically, a single A100 GPU—but observed that as training progressed, the loss actually increased, and the resulting images developed numerous voids. Upon inspection, I traced the issue to the target depth (the input depth appeared normal). Could you please help me analyze the underlying cause?
（我用相同的配置训练模型，1张A100 GPU，发现随着训练增加，反而loss变大，图像变得很多空洞，检查了下发现target depth，（输入的depth正常），可以帮我分析一下原因吗？）
input and image depths:
<img width="1030" height="1044" alt="Image" src="https://github.com/user-attachments/assets/b20482c2-2129-4f6d-83b6-f558431367a2" />

result:
<img width="998" height="1356" alt="Image" src="https://github.com/user-attachments/assets/5107cf79-332b-4b93-8649-200b894aa008" />

I used pytorch 2.6.0, Is this the likely reason?

**Comment by YCZ55** [comment]
The results for the first 200 steps were normal, but as training progressed, performance steadily deteriorated.
（前200步的结果正常，但是随着训练增加就越来越差了。）

**Comment by YCZ55** [comment]
loss is 

<img width="1502" height="902" alt="Image" src="https://github.com/user-attachments/assets/16ef26f0-8da9-4a93-b2ca-d1ef4fbb4f75" />

**Comment by hebing-sjtu** [comment]
Hi there, thanks for providing these details!

The issue you're experiencing—where the loss diverges, voids appear, and depth breaks down—typically happens when the 2D Gaussians shrink too much or become too transparent during optimization.

To fix this, you need to adjust your alpha thresholds depending on the phase:

https://github.com/hebing-sjtu/SurfSplat/blob/7ebe8901a881eedebe30b1ab591c032d29a61c2b/src/model/decoder/cuda_splatting.py#L128

During training: Set the alpha threshold to a larger value (like 0.1) . This prevents the Gaussians from fading out and maintains a solid geometric structure.

During inference/rendering: Remember to rewrite the alpha threshold back to a much smaller value to maintain correct and high-quality rendering.

We actually discuss these specific parameter adjustments in the experimental details section of our paper. Give that a try and let me know if it stabilizes your training!
